content/en/blog/falcosidekick-reponse-engine-part-6-cloud-run.md

```diff
@@ -0,0 +1,189 @@
+--set falcosidekick.config.gcp.cloudrun.jwt=$(gcloud auth print-identity-token)
+```
+
+Check the logs to see if `Cloud Run` output enabled for Falcosidekick.
+
+```shell
+$ kubectl logs -f falco-falcosidekick-7cd7bc6859-2nd9t --namespace falco
+falco-falcosidekick-7cd7bc6859-2nd9t falcosidekick 2021/06/07 16:03:14 [INFO]  : Enabled Outputs : [GCPCloudRun WebUI]
+falco-falcosidekick-7cd7bc6859-2nd9t falcosidekick 2021/06/07 16:03:14 [INFO]  : Falco Sidekick is up and listening on :2801
+...
+```
+
+If you see the `GCPCloudRun` in the list of enabled outputs, you consider that everything is good 👍.
+
+### Test
+
+We start by creating a dumb pod:
+
+```shell
+$ kubectl run alpine --namespace default --image=alpine --restart='Never' -- sh -c "sleep 600"
+pod/alpine created
+```
+
+```shell
+$ kubectl get pods --namespace default
+AME     READY   STATUS    RESTARTS   AGE
+alpine   1/1     Running   0          11s
+```
+
+Let's run a *shell* command inside and see what happens:
+
+```shell
+$ kubectl exec -i --tty alpine --namespace default -- sh -c "uptime"
+19:27:21 up 50 min,  load average: 0.11, 0.12, 0.11
+```
+
+As expected we got the result of our command, but, if we get the status of the pod we retrieve:
+
+```shell
+$ kubectl get pods --namespace default
+NAME     READY   STATUS        RESTARTS   AGE
+alpine   1/1     Terminating   0          103s
+```
+
+So, if you take a look at the logs of the `Cloud Run` function from the Google Cloud Console, you should see the following output:
+
+![cloud_run_function_output](/docs/images/cloud_run_function_outout.png)
+
+We can check the logs of the `Falco`, and `Falcosidekick` to see what happened.
+
+```shell
+ $ kubectl logs daemonset/falco --namespace falco
+..
+{"output":"19:27:21.002873265: Notice A shell was spawned in a container with an attached terminal (user=root user_loginuid=-1 k8s.ns=default k8s.pod=alpine container=97c9868ea832 shell=sh parent=runc cmdline=sh -c uptime terminal=34816 container_id=97c9868ea832 image=alpine) k8s.ns=default k8s.pod=alpine container=97c9868ea832","priority":"Notice","rule":"Terminal shell in container","time":"2021-04-10T19:27:21.002873265Z", "output_fields": {"container.id":"97c9868ea832","container.image.repository":"alpine","evt.time":1618082841002873265,"k8s.ns.name":"default","k8s.pod.name":"alpine","proc.cmdline":"sh -c uptime","proc.name":"sh","proc.pname":"runc","proc.tty":34816,"user.loginuid":-1,"user.name":"root"}}
+...
+```
+
+For `Falcosidekick`:
+
+```shell
+$ kubectl logs deployment/falco-falcosidekick --namespace falco
+2021/06/07 16:03:15 [INFO]  : GCPCloudRun - Post OK (200)
+2021/06/07 16:03:15 [INFO]  : GCPCloudRun - Function Response : OK
+..
+```
+
+## Conclusion
+
+We got another way to create a Response Engine with amazing pieces of software from Open Source world. Of course, it's just the beginning, feel free to share your functions and workflows with the community for starting the creation of a true library of remediation methods.
```

Comment:
We got another way to create a Response Engine with amazing pieces of software from the Open Source world. Of course, it's just the beginning, feel free to share your functions and workflows with the community for creating a library of remediation methods.

Reply:
thank you so much, done.